pkg/controllers/pod_controller.go

```diff
@@ -56,8 +65,9 @@ func (r *PodReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&corev1.Pod{}).
 		WithEventFilter(predicate.NewPredicateFuncs(func(object client.Object) bool {
 			pod, ok := object.(*corev1.Pod)
-			// Only reconcile pods that are part of JobSets using exclusive placement.
-			return ok && usingExclusivePlacement(pod)
+			// Only reconcile leader pods which have been scheduled which are part of
+			// JobSets using exclusive placement.
+			return ok && placement.IsLeaderPod(pod) && isScheduled(pod) && usingExclusivePlacement(pod)
```

Comment:
we should skip deleted pods too, we can do that by checking if the `DeletionTimestamp` is set.